AU2/test/plugins/util/custom_plugins/test_PolicePlugin.py

```diff
@@ -0,0 +1,73 @@
+import datetime
+
+from AU2.database.AssassinsDatabase import ASSASSINS_DATABASE
+from AU2.database.EventsDatabase import EVENTS_DATABASE
+from AU2.database.model import Event
+from AU2.plugins.custom_plugins.PolicePlugin import PolicePlugin
+from AU2.test.test_utils import MockGame, some_players, plugin_test, dummy_event
+
+
+class TestPolicePlugin:
+
+    @plugin_test
+    def test_resurrect_as_police(self):
+        """
+        Tests that Assassin -> Resurrect as Police
+        """
+        k = 50
+        n = 4*k
+        p = some_players(n)
+        game = MockGame().having_assassins(p)
+        plugin = PolicePlugin()
+
+        # add some regular deaths
+        for i in range(k):
+            game.assassin(p[i]).kills(p[i + k])
+        # add some police deaths
+        for i in range(2*k, 3*k):
+            game.assassin(p[i + k]).is_police()
+            game.assassin(p[i]).kills(p[i + k])
+
+        # hide some assassins
+        game.assassin(p[k-1]).with_accomplices(p[2*k-1], p[3*k-1], p[4*k-1]).are_hidden()
+
+        # check that the list of players that can be resurrected is correct
+        # (only the dead non-police, non-hidden assassins should be shown!)
+        assert set(plugin.gather_dead_non_police()) == {p[i] + " identifier" for i in range(k, 2*k-1)}
+
+        # check that players get resurrected correctly
+        ##############################################
+        new_pseudonym = "New Pseudonym"
+        old_assassin = ASSASSINS_DATABASE.get(p[k] + " identifier")
+        old_idents = set(ASSASSINS_DATABASE.get_identifiers(include_hidden=lambda _: True))
+        # the way identifiers are generated in test mode messes with cloning
```

Comment:
Since this has caused issues for both of us, maybe we need to rewrite how test mode treats assassins a bit.

Reply:
Yeah...
In this particular case the annoying thing is that if identifiers worked normally in test mode all the other tests would need to be rewritten. Luckily we don't have that many tests at the moment...